### PR TITLE
Fix tidbmonitor externallabels unrecognized environment variables. 

### LINKED ISSUE #3782: Prometheus configuration unrecognized environment variable
## Bug Report

**What did you do?**
I  deploy tidbmonitor and prometheus configuration unrecognized external labels environment variable.
image

**What did you expect to see?**
 Namespace and podName configuration replace with  current container  environment variable.
**What did you see instead?**
image

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Fix #3782 

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 

### Release Notes

```release-note
Fix tidbmonitor externallabels unrecognized environment variables bug. 
```

### CODE REVIEW

File: pkg/monitor/monitor/util.go [hunk 1]
```diff
@@ -467,16 +477,16 @@ func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.T
...
+		commands = append(commands, monitor.Spec.Prometheus.Config.CommandOptions...)
 	}
 	if monitor.Spec.Prometheus.DisableCompaction || monitor.Spec.Thanos != nil {
-		c.Command = append(c.Command, "--storage.tsdb.max-block-duration=2h")
-		c.Command = append(c.Command, "--storage.tsdb.min-block-duration=2h")
+		commands = append(commands, "--storage.tsdb.max-block-duration=2h")
+		commands = append(commands, "--storage.tsdb.min-block-duration=2h")
 	}
-
+	c.Command = append(c.Command, strings.Join(commands, " "))
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:489]: No need to `Join` here, right?

File: pkg/monitor/monitor/util.go [hunk 2]
```diff
@@ -423,17 +424,14 @@ func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMClu
 }
 
 func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster, dc *v1alpha1.DMCluster) core.Container {
+	commands := []string{"sed 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/prometheus.yml > /tmp/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/tmp/prometheus.yml --storage.tsdb.path=/data/prometheus " + fmt.Sprintf("--storage.tsdb.retention=%dd", monitor.Spec.Prometheus.ReserveDays)}
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:427]: How can we implement the feature that we can update the configuration of Prometheus dynamically if we put the configuration file to /tmp dir in the Prometheus container?

Comment (mikechengwei) [reply pkg/monitor/monitor/util.go]: After discussion, we put new generated configuration file  to `/etc/prometheus/config_out` folder and mount original configmap to new folder `/etc/prometheus/config`.

File: pkg/monitor/monitor/util.go [hunk 3]
```diff
@@ -423,17 +424,14 @@ func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMClu
 }
 
 func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster, dc *v1alpha1.DMCluster) core.Container {
+	commands := []string{"sed 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/config/prometheus.yml > /etc/prometheus/config_out/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/etc/prometheus/config_out/prometheus.yml --storage.tsdb.path=/data/prometheus " + fmt.Sprintf("--storage.tsdb.retention=%dd", monitor.Spec.Prometheus.ReserveDays)}
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:427]: Should we follow the suggestion [here](https://github.com/prometheus/prometheus/issues/2357#issuecomment-368601662)?

Comment (mikechengwei) [reply pkg/monitor/monitor/util.go:427]: I have follow this pr.

**Comment by mikechengwei** [comment]
image

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**